### ISSUE TITLE
Some special characters on request aren't accepted by tomcat

### ISSUE THREAD

**Issue by jaarandaz** [lbdremy/solr-node-client#197]
I'm using Tomcat ver 7.0.52.0 and it is failing when accesing to

/solr/unience/select?q=best&defType=dismax&qf=s_name^1.3%20s_shortName^1.4%20s_isin^1.5%20s_managementCompany^1.2&start=0&rows=10&wt=json

but it works when accessing to

/solr/unience/select?q=best&defType=dismax&qf=s_name%5E1.3%20s_shortName%5E1.4%20s_isin%5E1.5%20s_managementCompany%5E1.2&start=0&rows=10&wt=json

Is there any configuration option to be sure "^" becomes %5E on the request? Thank you

**Comment by nicolasembleton** [comment]
Yeah, it's the only characted that isn't being encoded because of this library: https://github.com/Gozala/querystring/blob/master/encode.js#L49
And that the '^' character is being used as the equal character for stringification: https://github.com/lbdremy/solr-node-client/blob/master/lib/query.js#L588

To have it %5E-ed, you'll need to replace that '^' everywhere in the code to '%5E'. I suppose that'd make sense too. 

On the other hand, I never saw anyone complaining about that either so I'm wondering why this version of Tomcat is behaving? Anyone had trouble like this one?

**Comment by josuehd** [comment]
yeah, I'm having the same issue using tomcat 7.0.76

**Comment by julianlam** [comment]
FWIW, still seeing this with tomcat 9.0.16-3~18.04.1 (Ubuntu)
